sn_transfers/src/transfers/signed_transaction.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright 2024 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use std::collections::BTreeSet;
+
+use crate::error::Result;
+use crate::{
+    CashNote, DerivationIndex, MainPubkey, MainSecretKey, NanoTokens, OutputPurpose, SignedSpend,
+    SpendReason, TransferError, UnsignedTransaction,
+};
+use serde::{Deserialize, Serialize};
+
+/// A local transaction that has been signed and is ready to be executed on the Network
+#[derive(custom_debug::Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct SignedTransaction {
+    /// Output CashNotes ready to be packaged into a `Transfer`
+    #[debug(skip)]
+    pub output_cashnotes: Vec<CashNote>,
+    /// Change CashNote ready to be added back to our wallet
+    #[debug(skip)]
+    pub change_cashnote: Option<CashNote>,
+    /// All the spends ready to be sent to the Network
+    pub spends: BTreeSet<SignedSpend>,
+}
+
+impl SignedTransaction {
+    /// Create a new `SignedTransaction`
+    pub fn new(
+        available_cash_notes: Vec<CashNote>,
+        recipients: Vec<(NanoTokens, MainPubkey, DerivationIndex, OutputPurpose)>,
+        change_to: MainPubkey,
+        input_reason_hash: SpendReason,
+        main_key: &MainSecretKey,
+    ) -> Result<Self> {
+        let unsigned_tx = UnsignedTransaction::new(
+            available_cash_notes,
+            recipients,
+            change_to,
+            input_reason_hash,
+        )?;
+        let signed_tx = unsigned_tx.sign(main_key)?;
+        Ok(signed_tx)
+    }
+
+    /// Verify the `SignedTransaction`
+    pub fn verify(&self) -> Result<()> {
+        for cn in self.output_cashnotes.iter() {
+            cn.verify()?;
+        }
+        if let Some(ref cn) = self.change_cashnote {
+            cn.verify()?;
+        }
+        for spend in self.spends.iter() {
+            spend.verify()?;
+        }
+        Ok(())
+    }
+
+    /// Create a new `SignedTransaction` from a hex string
+    pub fn from_hex(hex: &str) -> Result<Self> {
```

Comment:
we shall only require `hex` format for a `transfer`, but never for a `transaction` ?

Reply:
This is used for the `wo_wallet` which doesn't have a private key to sign the Tx but can nonetheless create a Tx. 
The `hex` format is used so that this wallet can export this unsigned_tx to be signed by a hot_wallet.
I think this was done so we can interoperate with hardware keys and such.